### PR TITLE
HAPP-1536: Immnz schedules UI

### PR DESCRIPTION
**Pull request by bsavoini-fw**

### CODE REVIEW

File: app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import ca.bc.gov.bchealth.compose.theme.white
+import ca.bc.gov.bchealth.ui.custom.DecorativeImage
+import ca.bc.gov.bchealth.ui.home.immunizationschedules.ImmunizationSchedulesViewModel.ImmunizationSchedulesItem
+
+@Composable
+fun ImmunizationSchedulesScreen(
+    uiList: List<ImmunizationSchedulesItem>,
+    onClickItem: (String) -> Unit
+) {
+    LazyColumn(
```
Comment (PINAKIN-KANSARA-EY) [inline app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt:36]: abstract this out in to private method

**Review by PINAKIN-KANSARA-EY — CHANGES_REQUESTED** [review]

File: app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesFragment.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        MyHealthTheme {
+            Scaffold(
+                topBar = {
+                    MyHealthToolbar(
+                        title = stringResource(id = R.string.immnz_schedules_title),
+                        navigationAction = ::popNavigation
+                    )
+                },
+                content = {
+                    Box(
```
[the remark this answers is not in the document]
Comment (bsavoini-fw) [reply app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesFragment.kt:34]: done

File: app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            }
+        }
+    )
+}
+
+@Composable
+private fun ImmunizationScheduleUI(item: ImmunizationSchedulesItem, onClickItem: (String) -> Unit) {
+    val url = stringResource(id = item.url)
+
+    Row(
```
[the remark this answers is not in the document]
Comment (bsavoini-fw) [reply app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt:57]: done

File: app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt (hunk 1 above)
Comment (bsavoini-fw) [reply app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt:36]: in this case the LazyColumn is already the only thing being called in this method, so it doesn't make sense to extract to a method

Comment (PINAKIN-KANSARA-EY) [reply app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt:36]: Hat will help you to put the viewModel call to get the list inside the screen block with launch effect, which you have to do eventually.

File: app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (bsavoini-fw) [reply app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt]: done

File: app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt (hunk 1 above)
Comment (bsavoini-fw) [reply app/src/main/java/ca/bc/gov/bchealth/ui/home/immunizationschedules/ImmunizationSchedulesScreen.kt:55]: done

**Review by PINAKIN-KANSARA-EY — APPROVED** [review]

**Pull request merged**